GHzDACs/ghz_fpga_server.py

```diff
@@ -990,7 +991,13 @@ def findDevices(self):
         for key in removals:
             bg = self.boardGroups[key]
             del self.boardGroups[key]
-            yield bg.shutdown()
+            try:
+                yield bg.shutdown()
+            except labrad.types.Error, e:
+                if 'no such server' in str(e):
```

Comment:
This kind of check that relies on the content of an error message is really brittle, and generally not something you want to rely on for correctness. I would suggest instead that we add a check in the shutdown method itself to see whether the server still exists and skip the expire_context call if not, and here just capture any exception and log it.

```python
# in BoardGroup:
@inlineCallbacks
def shutdown(self):
    cxn = self.directEthernetServer._cxn
    servers = yield cxn.manager.servers()
    server_ids = set(id for id, name in servers)
    if self.directEthernetServer.ID in ids:
        yield cxn.manager.expire_context(
                self.directEthernetServer.ID, context=self.ctx)

...
# here
for key in removals:
    bg = self.boardGroups[key]
    del self.boardGroups[key]
    try:
        yield bg.shutdown()
    except Exception, e:
        logging.error('Exception in BoardGroup shutdown: {}'.format(e))
```

One of the things that people have raised before is having a better way to deal with errors based on error codes, which would be much better than matching strings in error messages. That's something we could explore, but clearly outside the scope of this change.